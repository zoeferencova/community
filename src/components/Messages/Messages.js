import React, { Component } from "react";
import moment from "moment";
import tz from "moment-timezone";
import CommUnityContext from "../../contexts/context";
import styles from "../MessageLayout/MessageLayout.module.css";

export default class Messages extends Component {
    static contextType = CommUnityContext;

    constructor(props) {
        super(props)
        this.scrollDown = this.scrollDown.bind(this)
    }

    scrollDown() {
        const { container } = this.refs;
        container.scrollTop = container.scrollHeight;
    }

    componentDidMount() {
        this.scrollDown()
    }
    
    componentDidUpdate() {
        this.scrollDown()
    }

    makeMessages = () => {
        const { messages, user, receiver } = this.props;
        if (messages !== undefined) {
            return messages.map(msg => {
                const timezoneTimestamp = moment.tz(msg.message_timestamp, 'UTC').clone().tz(this.context.timeZone)
                return (
                    <div key={msg.id} className={`${styles.messageContainer} ${msg.sender_id === user.id && styles.right}`}>
                        <div className={styles.time}>{timezoneTimestamp.format("h:mm A")}</div>
                        <div className={styles.data}>
                            <div className={styles.message}>{msg.message_content}</div>
                            <div className={styles.name}>{msg.sender_id === user.id ? user.first_name : receiver.first_name}</div>
                        </div>
                    </div>
                )
            })
        }
    }

    makeTypingUsers = () => {
        const { typingUsers } = this.props;
        return typingUsers.map(u => {
            return (
                <div key={u.first_name} className={styles.typingUser}>
                    {`${u.first_name} is typing...`}
                </div>
            )
        })
    }
    
    render() {
        return (
            <div ref="container" className={styles.threadContainer}>
                <div className={styles.thread}>
                    {this.makeMessages()}
                    {/* {this.makeTypingUsers()} */}
                </div>
            </div>
        )
    }
}