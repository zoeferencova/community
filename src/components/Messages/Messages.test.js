import React from "react";
import { render } from "enzyme";
import Messages from "./Messages";
import CommUnityContext from "../../contexts/context";
import mockContext from "../../contexts/mock-context";

describe("Messages", () => {
    it("renders Messages component given mock context data", () => {
        const mockProps = {
            messages: [
                {
                    chat_id: 25,
                    id: 285,
                    message_content: "hi",
                    message_timestamp: "2020-09-17T18:22:47.088676",
                    sender_id: 1
                },
                {
                    chat_id: 25,
                    id: 286,
                    message_content: "hey",
                    message_timestamp: "2020-09-17T18:22:47.088676",
                    sender_id: 2
                },
            ],
            user: {
                email: "james@gmail.com",
                first_name: "James",
                id: 2,
                location: {lat: 40.748225, lng: -73.89698620000001},
                radius: "3.00"
            },
            receiver: {
                first_name: "James",
                id: 2
            }
        }
        
        const wrapper = render(
            <CommUnityContext.Provider value={mockContext}>
                <Messages {...mockProps} />
            </CommUnityContext.Provider>
        ); 

        expect(wrapper).toMatchSnapshot();
    });
});