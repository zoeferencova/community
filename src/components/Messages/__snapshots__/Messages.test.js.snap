// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages renders Messages component given mock context data 1`] = `
<div
  class="threadContainer"
>
  <div
    class="thread"
  >
    <div
      class="messageContainer false lastInThread"
    >
      <div
        class="circle indigo messagePic"
        first_name="James"
      >
        J
      </div>
      <div
        class="data"
      >
        <div
          class="message"
        >
          hi
        </div>
        <div
          class="time"
        >
          2:22 PM
        </div>
      </div>
    </div>
    <div
      class="messageContainer right lastInThread"
    >
      <div
        class="circle indigo messagePic"
        first_name="James"
      >
        J
      </div>
      <div
        class="data"
      >
        <div
          class="message"
        >
          hey
        </div>
        <div
          class="time"
        >
          2:22 PM
        </div>
      </div>
    </div>
  </div>
</div>
`;
